tinygrad/ops.py

```diff
@@ -62,7 +62,7 @@ class UOps(Enum):
 @dataclass(frozen=True, eq=False)
 class UOp:
   op: UOps
-  dtype: Optional[DType] = None
+  dtype: DType = dtypes.void
```

Comment:
does NOp work correctly?

Reply:
Appears to, had to handle a case where UPats are instantiated with NOps, but that's it